### ISSUE TITLE
mxBootstrap: parallel not working?

### ISSUE THREAD

**Issue by tbates** [OpenMx/OpenMx#181]
I'm getting a slightly longer runtime for a bootstrap with parallel on than with it off (on a machine with 4 real cores)

```R
m3_boot_serial = mxBootstrap(m3, replications=1000, parallel=TRUE)
m3_boot_parallel = mxBootstrap(m3, replications=1000, parallel=FALSE)
umx_time(c(m3_boot_serial, m3_boot_parallel))
crime_model: 02.83 seconds, 
crime_model: 02.78 seconds(∆: -0.049), 98.284%
```

**Comment by jpritikin** [comment]
Yeah, it was never implemented. You do get benefits from parallel if the multiple cores help within a single model. However, it would be more useful to fit each bootstrap replication in parallel.

**Comment by mcneale** [comment]
So it seems... any reason you reverse-coded the model names?

I agree that large granularity usually works best in parallel - the exception being when you have more   processors than granules.

**Comment by tbates** [comment]
fixed the reverse naming.
I guess we should print `"polite note: parallel is not yet implemented"`

I'll update the Rd to  say not parallel not working and note that algebras are not booted.

**Comment by mcneale** [comment]
parallel _bootstrap_ is not yet implemented.  Parallel's a big boon elsewhere in OpenMx.

**Comment by tbates** [comment]
Updated Rd to note that `parallel` is (currently) ignored.

I tested our boot speed as-is/1-core: We show a 5.7x speed-up compared to `lavaan(..., se = "bootstrap")`, and it is linear from 100 to > 1000 bootstraps, So that at least validates the Clang/CRAN backend hassles are worth it! Roll-on multicore boot!

**Comment by mcneale** [comment]
Isn't it the case that FIML analyses will use multiple cores  anyway?   Yes, a bit less efficient than running bigger  granules like entire  model to be bootstrapped, but better than nothing for FIML.  Speed improvement might not be that much switching in  these cases - though it would be for non-FIML.

Cheers
Mike

Michael C. Neale, Ph.D.
Professor, Virginia Institute for Psychiatric and Behavioral GeneticsIPBG
Virginia Commonwealth University
Shipping Address: 800 East Leigh St. Suite 1-156, Richmond VA 23219-1534
Mailing Address : Box 980126 MCV Richmond VA 23298-0126
Telephone: (804) 828-3369  Email: michael.neale@vcuhealth.org
________________________________

**Comment by jpritikin** [comment]
> Isn't it the case that FIML analyses will use multiple cores anyway?

Yes, but not all models require FIML.
